control-base/src/main/java/org/oskari/control/layer/DescribeLayerHandler.java

```diff
@@ -102,53 +100,83 @@ private LayerExtendedOutput getLayerDetails(ActionParameters params, OskariLayer
+        WFSLayerOptions opts = new WFSLayerOptions(layer.getOptions());
+        output.properties = getProperties(caps, attr, lang);
+        output.data = getData(caps, attr, opts);
+    }
+
+    private List<FeatureProperties> getProperties(LayerCapabilitiesWFS caps, WFSLayerAttributes attr , String lang) {
         List<FeatureProperties> props = new ArrayList<>();
 
-        JSONObject locale = getPropertiesLocale(layer);
+        List<String> selected = attr.getSelectedAttributes(lang);
+        Optional<JSONObject> locale = attr.getLocalization(lang);
+        JSONObject format = attr.getAttributesData().optJSONObject("format");
+
         caps.getFeatureProperties().stream().forEach(prop -> {
             FeatureProperties p = new FeatureProperties();
             p.name = prop.name;
             p.type = WFSConversionHelper.getSimpleType(prop.type);
             p.rawType = prop.type;
-            p.label = getLabelForProperty(locale, prop.name, lang);
+            p.label = locale.isPresent() ? locale.get().optString(prop.name, null) : null;
+            p.format = getPropertyFormat(format, prop.name);
+            if (!selected.isEmpty()) {
+                int index = selected.indexOf(p.name);
+                if (index == -1) {
+                    p.hidden = true;
+                    p.order = selected.size();
+                } else {
+                    p.order = index;
+                }
+            }
             props.add(p);
         });
+        if (!selected.isEmpty()) {
+            props.sort(Comparator.comparingInt(a -> a.order));
+        }
         return props;
     }
-    private JSONObject getPropertiesLocale(OskariLayer layer) {
-        JSONObject attrs = layer.getAttributes();
-        if (attrs == null) {
-            return null;
-        }
-        JSONObject data = attrs.optJSONObject("data");
-        if (data == null) {
+    private Map<String, Object> getPropertyFormat (JSONObject format, String name) {
+        if (format == null) {
             return null;
         }
-        return data.optJSONObject("locale");
+        return JSONHelper.getObjectAsMap(format.optJSONObject(name));
     }
 
-    private String getLabelForProperty(JSONObject locale, String prop, String lang) {
-        if (locale == null) {
-            return null;
-        }
-        JSONObject labelsForLang = locale.optJSONObject(lang);
-        if (labelsForLang == null) {
-            return null;
+    private Map<String, Object> getData (LayerCapabilitiesWFS caps, WFSLayerAttributes attr, WFSLayerOptions opts) {
+        Map<String, Object> data = new HashMap<>();
+
+        JSONObject attrData = attr.getAttributesData();
+        data.put(WFSLayerAttributes.KEY_NO_DATA_VALUE, attr.getNoDataValue());
+        data.put(WFSLayerAttributes.KEY_COMMON_ID, attr.getCommonId());
+        data.put(WFSLayerAttributes.KEY_WPS_TYPE, attrData.optString(WFSLayerAttributes.KEY_WPS_TYPE, null));
```

Comment:
I think we should skip/not handle the WPS_TYPE for now since it's only for analysislayers and DescribeLayer only works with layers in oskari_maplayer table currently.

Reply:
removed